src/main/java/com/fasterxml/jackson/databind/ObjectReader.java

```diff
@@ -355,7 +355,11 @@ protected JsonToken _initForReading(DeserializationContext ctxt, JsonParser p)
             t = p.nextToken();
             if (t == null) {
                 // Throw mapping exception, since it's failure to map, not an actual parsing problem
-                ctxt.reportMissingContent(null); // default msg is fine
+                if (this._config.hasDeserializationFeatures(DeserializationFeature.READ_NULL_OR_MISSING_CONTAINER_AS_EMPTY.getMask())) {
```

Comment:
This is semantically different, as it is actually missing the whole JSON content (which makes it invalid; empty content is not a valid JSON value), so I don't think it is right to map it to anything. If such handling of invalid content is desired it should be fully separate feature, especially as it would apply to all content. I am not necessarily against such a feature but it belongs to a different issue/patch.